Fix error throwing

ThomWright opened the pull request:
An Error was being created, then overwritten with an object

This causes problems for clients that expect real Errors in
promise rejections

ThomWright committed:
Fix error throwing

An Error was being created, then overwritten with an object

This causes problems for clients that expect real Errors in
promise rejections

ThomWright committed:
Add errors messages and extra error code

The message property on Errors is very useful in logs

The 23505 error can be thrown when trying to create
the same database concurrently (whoops :D)

olalonde commented on index.js:
```diff
@@ -35,11 +45,9 @@ function createOrDropDatabase(action) {
       client.query(sql, function (pgErr, res) {
         var err;
         if (pgErr) {
-          err = new Error();
```
oops, that line was useless I guess

olalonde commented:
LGTM, though it would be nice to encapsulate the error logic into a custom Error class (e.g. `PgError`) so that

```javascript
     err = new Error(errors[pgErr.code].message || "Unknown Postgres error");
     err.name = errors[pgErr.code].name || 'PostgresError',
     err.pgErr = pgErr
```

becomes

```javascript
   err = new PgError(pgErr)
```

ThomWright committed:
Use custom error type and improve messaging

The custom error now includes the message from the underlying
pg error - useful for debugging

olalonde commented:
Thanks @ThomWright ! @bcoe want to merge?

bcoe commented:
@olalonde :shipit: \o/

bcoe commented:
@ThomWright thanks for the contribution.

bcoe commented:
@ThomWright update published in `+ pgtools@0.2.3` 👍
